configure.ac

```diff
@@ -3790,13 +3790,23 @@ AS_IF([test "$cross_compiling" = no],
     )
 )
 
+dnl Needs newer objcopy in order to localize symbols added by rustc
+YJIT_OBJCOPY_OK=no
+AS_CASE(["$OBJCOPY"], [""|:], [], [
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[]])], [
+        AS_IF([$OBJCOPY --wildcard --keep-global-symbol='*main' --localize-symbol='__*' conftest.$OBJEXT 2>/dev/null], [
+            YJIT_OBJCOPY_OK=yes
+        ])
+    ])
+])
+
 dnl build YJIT in release mode if rustc >= 1.58.0 is present and we are on a supported platform
 AC_ARG_ENABLE(yjit,
     AS_HELP_STRING([--enable-yjit],
     [enable in-process JIT compiler that requires Rust build tools. enabled by default on supported platforms if rustc 1.58.0+ is available]),
     [YJIT_SUPPORT=$enableval],
-    [AS_CASE(["$YJIT_TARGET_OK:$YJIT_RUSTC_OK"],
-        [yes:yes], [
+    [AS_CASE(["$YJIT_TARGET_OK:$YJIT_RUSTC_OK:$YJIT_OBJCOPY_OK"],
```

Comment:
@nobu https://github.com/ruby/ruby/pull/7115 used objcopy only for ELF platforms (@XrXr Let me know if I'm wrong), so this check should be skipped for darwin. hsbt temporarily reverted this PR https://github.com/ruby/ruby/commit/a642a94b68a31c0e7c092087114c269132cdb159 since it now disables YJIT build on macOS by default.

Reply:
Is it for all ELF platforms, not only Linux?

Reply:
ELF platforms other than Linux, is really just FreeBSD, and it has a new enough objcopy (I couldn't test on OpenBSD). YJIT on macOS doesn't need objcopy at all because the linker has options for hiding everything.